testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,693 @@
+                            }
+                        } catch (NullPointerException npex) {
+                            npex.printStackTrace();
+                        }
+                        if (threadLocal) {
+                            Object tmp = fieldList[i].get(null);
+                            ((ThreadLocal)tmp).set(ob_0);
+                            ob_0 = tmp;
+                        }
+
+                        FieldUtils.writeField(fieldList[i], (Object)null, ob_0, true);
+
+                        String output = fieldName + " set\n";
+                        Files.write(Paths.get(reflectionFile), output.getBytes(),
+                                StandardOpenOption.APPEND);
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                        String outputNormalError = fieldName + " reflectionError: " + e + "\n";
+                        Files.write(Paths.get(reflectionFile), outputNormalError.getBytes(),
+                                StandardOpenOption.APPEND);
+                    }
+                }
+            } catch (Exception e) {
+            }
+        } catch (Exception e) {
+            String output = fieldName + " deserializeError: " + e + "\n";
+            Files.write(Paths.get(reflectionFile), output.getBytes(),
+                    StandardOpenOption.APPEND);
+        }
+    }
+
+    public void fixingFieldList(List<String> fields) throws IOException {
+        for (int index = 0; index < fields.size(); index++) {
+            String fieldName = fields.get(index);
+            fixing(fieldName);
+        }
+    }
+
+    public void capture() {
+        try {
+            String state = Configuration.config().getProperty("statecapture.state");
+            if (!state.equals("double_victim")) {
+                capture_real();
+            } else {
+                capture_class();
+            }
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Adds the current serialized reachable state to the currentTestStates list
+     * and the current roots to the currentRoots list.
+     * @throws IOException
+     */
+    public void capture_real() throws IOException {
+        String phase = Configuration.config().getProperty("statecapture.phase");
+        PrintWriter writer;
+
+        // read whitelist;
+        try (BufferedReader br = new BufferedReader(new FileReader(Configuration.config().getProperty("replay.pkgFile")))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                whiteList.add(line);
+            }
+        }
+        catch (Exception e) {
+            return;
+        }
+
+        String subxmlDir = createSubxmlFold();
+
+        Set<String> allFieldName = new HashSet<String>();
+        Class[] loadedClasses = MainAgent.getInstrumentation().getAllLoadedClasses();
+        File file = new File(rootFold + "/eagerLoadingFields.txt");
```

Comment:
Instead of checking whether this file exists, can it just be passed in as an input (e.g., through some new property)? This would be to prevent some run from accidentally going through this mode if the file happens to exist there.